src/intersperse.js

```diff
@@ -0,0 +1,31 @@
+var _checkForMethod = require('./internal/_checkForMethod');
+var _curry2 = require('./internal/_curry2');
+
+
+/**
+ * Creates a new list with the separator interposed between elements.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig * -> [a] -> [a]
+ * @param {*} separator The element to add to the list.
+ * @param {Array} list The list to be interposed.
+ * @return {Array} The new list.
+ * @example
+ *
+ *      R.interpserse('n', ['ba', 'a', 'a']); //=> ['ba', 'n', 'a', 'n', 'a']
+ */
+module.exports = _curry2(_checkForMethod('intersperse', function intersperse(separator, list) {
+  var out = [];
+  var idx = -1;
+  var length = list.length;
+  while (++idx < length) {
+    if (idx === length - 1) {
+      out = out.concat(list[idx]);
+    } else {
+      out = out.concat(list[idx], separator);
```

Comment:
Push is just better in this case David
On Apr 21, 2015 9:51 PM, "David Chambers" <notifications@github.com> wrote:

> In src/intersperse.js
> <https://github.com/ramda/ramda/pull/1043#discussion_r28838931>:
>
> > + * @param {*} separator The element to add to the list.
> > + * @param {Array} list The list to be interposed.
> > + * @return {Array} The new list.
> > + * @example
> > + *
> > + *      R.interpserse('n', ['ba', 'a', 'a']); //=> ['ba', 'n', 'a', 'n', 'a']
> > + */
> > +module.exports = _curry2(_checkForMethod('intersperse', function intersperse(separator, list) {
> > +  var out = [];
> > +  var idx = -1;
> > +  var length = list.length;
> > +  while (++idx < length) {
> > +    if (idx === length - 1) {
> > +      out = out.concat(list[idx]);
> > +    } else {
> > +      out = out.concat(list[idx], separator);
>
> Didn't we decide to use xs[xs.length] = x? See #891
> <https://github.com/ramda/ramda/pull/891>.
>
> I'd be happy for us to reverse that decision, but were we to do so it
> would be orthogonal to this change.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ramda/ramda/pull/1043/files#r28838931>.
>